core.coffee

```diff
@@ -36,6 +36,59 @@
 		'Options without $ prefixes (e.g. `options: filter: ...`) are deprecated, please use a $ prefix (eg `options: $filter: ...`).'
 	)
 
+	class Poll
+		channels:
+			'error': []
+			'data': []
+
+		stopped: false
+
+		intervalTime: 10000
+
+		constructor: (requestFn, intervalTime) ->
+			if typeof intervalTime is 'number'
+				@intervalTime = intervalTime
+
+			@requestFn =  requestFn
+			@start()
+
+		runRequest: ->
+			return if @stopped
+			@requestFn()
+			.then (response) =>
+				return if @stopped
+
+				fn(response) for fn in @channels.data
+			.catch (err) =>
+				return if @stopped
+
+				fn(err) for fn in @channels.error
```

Comment:
Having the `for` as a the last statement will collect up all of the return values of the handlers and return them, which adds unnecessary overhead